php/class-customize-snapshot-manager.php

```diff
@@ -103,23 +103,11 @@ public function __construct( Plugin $plugin ) {
 
 		$this->snapshot = new Customize_Snapshot( $this, $uuid, $apply_dirty );
 
-		// Set the return URL in the Customizer.
-		if ( $this->snapshot->is_preview() ) {
-			$that = $this;
-			add_action( 'customize_controls_init', function () use ( $uuid, $scope, $that ) {
-				$args = array(
-					'customize_snapshot_uuid' => $uuid,
-					'scope' => $scope,
-				);
-				$return_url = add_query_arg( $args, $that->customize_manager->get_return_url() );
-				$that->customize_manager->set_return_url( $return_url );
-			} );
-		}
-
+		add_action( 'customize_controls_init', array( $this, 'set_return_url' ) );
 		add_action( 'init', array( $this, 'maybe_force_redirect' ), 0 );
 		add_action( 'init', array( $this, 'create_post_type' ), 0 );
 		add_action( 'customize_controls_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
-		add_action( 'wp_ajax_customize_save', array( $this, 'set_snapshot_uuid' ) );
+		add_action( 'wp_ajax_customize_save', array( $this, 'set_snapshot_uuid' ), 0 );
```

Comment:
Yes that could work, though `set_snapshot_uuid` would need to be modified to filter `customize_save_response` instead of retuning `wp_send_json_error`.